### PR TITLE
JBIDE-26558 - Update WildFly to version 17 in examples.itests

### PR DESCRIPTION
**Pull request by zcervink**
JBIDE-26558 - Update WildFly to version 17 in examples.itests.

Signed-off-by: Zbynek Cervinka <zcervink@redhat.com>

**Jenkins:** https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/examples.wildfly.it.weekly/545/
**Jira:** https://issues.jboss.org/browse/JBIDE-26558

### CODE REVIEW

File: tests/org.jboss.tools.examples.ui.bot.test/pom.xml [hunk 1]
```diff
@@ -16,16 +16,16 @@
...
 		<!-- -DimportTestDefinition=${importTestDefinition}</systemProperties> -->
 		<systemProperties>${integrationTestsSystemProperties} -DexamplesLocation=${project.build.directory}/${examplesFolder} -Dmaven.settings.path=${maven.settings.path} -Drd.config=${reddeer.config} -DspecificQuickstarts=${specificQuickstarts} -DdeployOnServer=${deployOnServer}</systemProperties>
-		<quickstartsURLWildFly>https://github.com/wildfly/quickstart/archive/14.0.1.Final.zip</quickstartsURLWildFly>
+		<quickstartsURLWildFly>https://github.com/wildfly/quickstart/archive/17.0.0.Final.zip</quickstartsURLWildFly>
 		<quickstartsURLEAP>http://download.eng.brq.redhat.com/released/JBoss-middleware/eap7/7.2.1/jboss-eap-7.2.1-quickstarts.zip</quickstartsURLEAP>
 		<quickstartsURLJavaEE>https://github.com/javaee-samples/javaee7-samples/archive/master.zip</quickstartsURLJavaEE>
 		<maven.settings.path>${project.build.directory}/classes/settings.xml</maven.settings.path>
-		<reddeer.config.wildfly14>${project.build.directory}/classes/servers/wildfly-14.json</reddeer.config.wildfly14>
+		<reddeer.config.wildfly17>${project.build.directory}/classes/servers/wildfly-17.json</reddeer.config.wildfly17>
```
Comment (jkopriva) [inline tests/org.jboss.tools.examples.ui.bot.test/pom.xml]: You can change name of variables in pom to just WildFly(without version number), so you don't need to update it on so many places when updating WildFly.

File: tests/org.jboss.tools.examples.ui.bot.test/resources/servers/wildfly-blacklist-test-errors-regexes.json [hunk 2]
```diff
@@ -27,5 +27,73 @@
 		".*Element 'value' must have no element.*",
 		".*Value 'param1 flow1 value' is not facet-valid with respect to pattern.*",
 		".*Invalid content was found starting with elemen 'flow-call'.*"
+	],
+	"spring-greeter":[
+		".*Can not find the tag library descriptor for \"http://java.sun.com/jsp/jstl/core\".*",
+		".*Missing artifact org.apache.taglibs:taglibs-standard-impl:jar:1.2.6-RC1.*",
+		".*Missing artifact org.apache.taglibs:taglibs-standard-spec:jar:1.2.6-RC1.*",
+		".*Plugin execution not covered by lifecycle configuration: org.apache.maven.plugins:maven-checkstyle-plugin:3.0.0:checkstyle.*",
```
Comment (jkopriva) [inline tests/org.jboss.tools.examples.ui.bot.test/resources/servers/wildfly-blacklist-test-errors-regexes.json]: I suppose this plugin can be installed via target platform configurator(see https://github.com/zcervink/jbosstools-integration-tests/blob/JBIDE-26558/tests/org.jboss.tools.examples.ui.bot.test/pom.xml#L87) as it has been done with different check-style plugin(https://github.com/zcervink/jbosstools-integration-tests/blob/JBIDE-26558/tests/org.jboss.tools.examples.ui.bot.test/pom.xml#L98).

File: tests/org.jboss.tools.examples.ui.bot.test/pom.xml (hunk 1 above)
Comment (zcervink) [reply tests/org.jboss.tools.examples.ui.bot.test/pom.xml]: Fixed.

File: tests/org.jboss.tools.examples.ui.bot.test/resources/servers/wildfly-blacklist-test-errors-regexes.json (hunk 2 above)
Comment (zcervink) [reply tests/org.jboss.tools.examples.ui.bot.test/resources/servers/wildfly-blacklist-test-errors-regexes.json]: Fixed.

**Comment by zcervink** [comment]
Update on this issue - all comments fixed and tested on Jenkins: https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/examples.wildfly.it.weekly/549/

**Comment by zcervink** [comment]
@jkopriva Update on this issue - all issues fixed and tested - https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/examples.wildfly.it.weekly/552/

**Comment by jkopriva** [comment]
@zcervink you also need to update readme, because you have changed profile names.

**Comment by zcervink** [comment]
@jkopriva README updated.

**Comment by jkopriva** [comment]
@zcervink thanks, I have the last two notes:
- There are 3 failures like "There is no project with name " for projects: dist, wicket-ear and helloworld-classfiletransformer - do you have a JIRA for this issue?
- Have you reviewed files wildfly-blacklist-test-errors-regexes.json and wildfly-blacklist, if these files needs update? You have added new records in wildfly-blacklist-test-errors-regexes.json, but are the old records ok?

**Comment by zcervink** [comment]
> @zcervink thanks, I have the last two notes:
> 
> * There are 3 failures like "There is no project with name " for projects: dist, wicket-ear and helloworld-classfiletransformer - do you have a JIRA for this issue?

@jkopriva I have created the JIRA, here is the link - https://issues.jboss.org/browse/JBIDE-26758

**Comment by zcervink** [comment]
testPR

**Comment by zcervink** [comment]
> @zcervink thanks, I have the last two notes:
> 
> * Have you reviewed files wildfly-blacklist-test-errors-regexes.json and wildfly-blacklist, if these files needs update? You have added new records in wildfly-blacklist-test-errors-regexes.json, but are the old records ok?

@jkopriva I have updated the wildfly-blacklist-test-errors-regexes.json file. Are we going to merge now?

**Comment by jkopriva** [comment]
@zcervink I am looking at the test results and **jts** quickstart should be green, because it is blacklist regexes, but it is not, can you please look at it? Thank you!

**Comment by zcervink** [comment]
@jkopriva Sorry for that, regex file has been updated (there was no new issue, this issue has been reported several days ago, I just have not blacklisted it). PR has been also rebased.

**Comment by zcervink** [comment]
**Jenkins:** https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/examples.wildfly.it.weekly/559/

**Comment by zcervink** [comment]
testPR

**Comment by zcervink** [comment]
@jkopriva Everything fixed (except 3 missing projects and Exception during sync execution in UI thread): https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/examples.wildfly.it.weekly/564/

File: tests/org.jboss.tools.examples.ui.bot.test/resources/servers/wildfly-blacklist-test-errors-regexes.json [hunk 3]
```diff
@@ -1,11 +1,8 @@
 {
 	"jts": [
 		".*Invalid content was found starting with element 'iiop:ejb-name'.*",
-		".*Referenced file contains errors.*jboss-ejb3-spec-2_0\\.xsd.*"
-	],
-	"ejb-security-interceptors": [
-		".*Invalid content was found starting with element 'jee:interceptor-binding'.*",
-		".*Referenced file contains errors.*jboss-ejb3-spec-2_0\\.xsd.*"
+		".*Referenced file contains errors.*jboss-ejb3-spec-2_0\\.xsd.*",
+		".*cvc-complex-type.2.4.a: Invalid content was found starting with element 'iiop:binding-name'. One of '.\"urn:iiop\":ejb-name.' is expected.*"
```
Comment (jkopriva) [inline tests/org.jboss.tools.examples.ui.bot.test/resources/servers/wildfly-blacklist-test-errors-regexes.json]: Row 3 and 5 are quite similar, do you need them both?

File: tests/org.jboss.tools.examples.ui.bot.test/resources/servers/wildfly-blacklist-test-errors-regexes.json [hunk 4]
```diff
@@ -21,11 +18,71 @@
 		".*The import org.jboss.quickstarts.ws.jaxws.samples.retail.profile.DiscountRequest cannot be resolved.*",
 		".*The import org.jboss.quickstarts.ws.jaxws.samples.retail.profile.DiscountResponse cannot be resolved.*",
 		".*The import org.jboss.quickstarts.ws.jaxws.samples.retail.profile.ProfileMgmt cannot be resolved.*",
-		".*Type mismatch: cannot convert from ProfileMgmt to ProfileMgmt.*"
+		".*Type mismatch: cannot convert from ProfileMgmt to ProfileMgmt.*",
```
Comment (jkopriva) [inline tests/org.jboss.tools.examples.ui.bot.test/resources/servers/wildfly-blacklist-test-errors-regexes.json]: I have imported jaxws-retail in my workspace and previous records are not relevant(ot least on JDK11), can you please also review it and delete it if needed? Thank you!

File: tests/org.jboss.tools.examples.ui.bot.test/resources/servers/wildfly-blacklist-test-errors-regexes.json (hunk 3 above)
Comment (zcervink) [reply tests/org.jboss.tools.examples.ui.bot.test/resources/servers/wildfly-blacklist-test-errors-regexes.json]: Replaced by 1 line.

File: tests/org.jboss.tools.examples.ui.bot.test/resources/servers/wildfly-blacklist-test-errors-regexes.json (hunk 4 above)
Comment (zcervink) [reply tests/org.jboss.tools.examples.ui.bot.test/resources/servers/wildfly-blacklist-test-errors-regexes.json]: Fixed.

**Comment by zcervink** [comment]
**Jenkins:** https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/examples.wildfly.it.weekly/571/

**Comment by jkopriva** [comment]
LGTM, merged. Thanks!

**Pull request merged**